### DISCUSSION TITLE
Inconsistent accessibility of ValidationBehavior and BaseBehavior<TView> classes.

### DISCUSSION THREAD

**Discussion by MaximG1234** [CommunityToolkit/Maui#844] [New Feature Discussions]
Currently `ValidationBehavior` and `BaseBehavior<TView>` is declared as a public class which implies to me that it is 'encouraged' to use this as a starting point for your own Behaviours. Yet a number of key methods are defined as `internal` specifically:

```
BaseBehavior<TView>.TrySetBindingContext(Binding binding)
BaseBehavior<TView>.TryRemoveBindingContext()
ValidationBehavior.ValidateNestedAsync(CancellationToken token)
```

I wanted to create a `ConditionalMultiValidationBehavior` based on the MultiValidationBehavior where I would leverage attached properties to create AND OR conditions for a generic collection of behaviors but because the above methods are internal I am stuck.

IMHO either these entire classes should be marked as internal (which they obviously shouldnt) or all the underlying methods should be accessible. From an extremely cursory review I cannot see a strong reason why these methods have been marked as internal anyways.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
